shacl12-rules/index.html

```diff
@@ -728,40 +714,135 @@ <h4>Compact Syntax Abbreviations</h4>
+            <var>μ</var>, is a partial function <code><var>μ</var> : V → T</code>, 
+            where <var>V</var> is the set of all variables and <var>T</var> is the set of all RDF terms.
+            The domain of <var>μ</var> is denoted by <code><var>dom</var>(<var>μ</var>)</code>,
+            and it is the subset of <var>V</var> for which <var>μ</var> is defined.
+            We use the term [=solution=] where it is clear that a [=solution mapping=] is meant.
+            Write <var>μ<sub>0</sub></var> for the solution mapping such that
+            <code><var>dom</var>(<var>μ<sub>0</sub></var>)</code> is the empty set.
+          </dd>
 
+          <dt><dfn data-lt="substitution">substitution function</dfn></dt>
+          <dd>
+            A <a>substitution function</a>, or just <i>substitution</i>,
+            is a function <code>subst(<var>μ</var>, [=triple pattern=])</code> that returns a triple pattern
+            where each occurrence in the triple pattern of a variable that is in the <code><var>dom</var>(<var>μ</var>)</code>
+            is replaced by the [=RDF Term=] given by the [=solution mapping=] for <var>var</var>.
+            If the triple pattern result has no variables, then it is an [=RDF Triple=].
+          </dd>
+          
+          <dt><dfn>Evaluation Graph</dfn></dt>
+          <dd>
+            A [=evaluation graph=] is an [=RDF Graph=] that combines the base graph 
+            and all triples produce by rule during the the evaluation of a rule set.
+          </dd>
+
+          <dt><dfn data-dt="match">triple pattern match</dfn></dt>
+          <dd>
+              A [=triple pattern match=] finds the ways to 
+              map a triple pattern onto triples in an [=RDF Graph=].
+              <p>
+                Let <var>G</var> be an [=RDF graph=] and <var>TP</var> be a <a>triple pattern</a>.
+                The function `match(G, TP)` returns a set of all possible 
+                <a data-cite="shacl12-core#solutions">solutions</a> that, 
+                when applied to the triple pattern, produce a triple that is in the 
+                [=evaluation graph=]
+              </p>
+            <div class="def">
+              <p>
+                Let <var>G</var> be an [=RDF graph=] and <var>TP</var> be a <a>triple pattern</a>.
+              </p>
+              <pre>match(G, TP) = { <var>μ</var> | subst(<var>μ</var>, TP) is a triple in G }</pre>
+            </div>
+          </dd>
+
+          <dt><dfn data-lt="compatible">solution compatible</dfn></dt>
+          <dd>
+            Two solutions S1 and S2 are [=compatible=] if they agree on the variables in common.
+            <div class="def">
+              <p>
+              Let S1 and S2 be solutions.
+              </p>
+              <pre>compatible(<var>μ</var><sub>1</sub>, <var>μ</var><sub>2</sub>) = true
+                      if forall v in dom(μ1) intersection dom(μ2)
```

Comment:
"Forall" (with no space) is a programming-language-specific function. "For all" (with space) is English. These blocks appear to be an English narration, which calls for "for all". If these are blocks of some programming language, they all need to be reviewed with an eye to that vocabulary, so there must be a declaration of what that language is.
```suggestion
                      if for all v in dom(μ1) intersection dom(μ2)
```

Reply:
It is "programming language" in a general sense as is used in formal descriptions, including academic papers.
Later, we can replace `forall` with `∀`. Or not. This draft PR is work-in-progress.

Reply:
Hm. I would prefer `for all` (and `for each`), but I won't block on this, especially if the overall WG prefers `forall` (and `foreach`).

Reply:
`for each` is different - it's `for` loop.

It could be `forall X in ...`.

This PR is someway from being "finished". I put it up to show the direction.